plugins/inputs/azure_monitor/azure_monitor.go

```diff
@@ -0,0 +1,157 @@
+
+func (e *AzureMonitorError) Error() string {
+	return e.message
+}
+
+type AzureMonitorResponseValueName struct {
+	LocalizedValue string
+	Value          string
+}
+
+type AzureMonitorResponseTimeSeriesDatum struct {
+	Average   float64
+	TimeStamp string
+}
+
+type AzureMonitorResponseTimeSeries struct {
+	Data           []AzureMonitorResponseTimeSeriesDatum
+	MetadataValues []map[string]interface{}
+}
+
+type AzureMonitorResponseValue struct {
+	DisplayDescription string
+	ErrorCode          string
+	Id                 string
+	Name               AzureMonitorResponseValueName
+	// ResourceGroup string
+	TimeSeries []AzureMonitorResponseTimeSeries
+	Type       string
+	Unit       string
+}
+
+type AzureMonitorResponse struct {
+	Cost           float64
+	Interval       string
+	Namespace      string
+	ResourceRegion string
+	Timespan       string
+	Value          []AzureMonitorResponseValue
+}
+
+func parseResponse(resp *http.Response) (AzureMonitorResponse, error) {
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return AzureMonitorResponse{}, err
+	}
+
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
+		return AzureMonitorResponse{}, &AzureMonitorError{message: fmt.Sprintf("Azure monitor request returned error. Status %v:\n%v", resp.StatusCode, string(body))}
+	}
+
+	var response AzureMonitorResponse
+	decoder := json.NewDecoder(strings.NewReader(string(body)))
+	err = decoder.Decode(&response)
+	if err != nil {
+		return AzureMonitorResponse{}, err
+	}
+
+	return response, nil
+}
+
+type AzureMonitor struct {
+	ResourceId string              `toml:"resource_id"`
+	authorizer autorest.Authorizer `toml:"-"`
+	Log        telegraf.Logger     `toml:"-"`
+}
+
+func (a *AzureMonitor) Description() string {
+	return "Gather Azure monitor metrics"
+}
+
+func (a *AzureMonitor) SampleConfig() string {
+	return `
+  ## The Azure Resource ID for which metrics will be gathered
+  ##   ex: resource_id = "/subscriptions/<subscription_id>/resourceGroups/<resource_group>/providers/Microsoft.Storage/storageAccounts/<storage_account>"
+  # resource_id = ""
+	`
+}
+
+func (a *AzureMonitor) Init() error {
+	if a.ResourceId == "" {
+		return errors.New("resource_id must be configured")
+	}
+
+	authorizer, err := auth.NewAuthorizerFromEnvironment()
+	if err != nil {
+		return err
+	}
+	a.authorizer = authorizer
+	return nil
+}
+
+func (a *AzureMonitor) makeRequest() (*http.Response, error) {
+	client := http.Client{}
+
+	url := fmt.Sprintf("https://management.azure.com/%v/providers/microsoft.insights/metricDefinitions?api-version=2018-01-01", a.ResourceId)
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+	req, err = autorest.CreatePreparer(a.authorizer.WithAuthorization()).Prepare(req)
+	if err != nil {
+		return nil, err
+	}
+	return client.Do(req)
+}
+
+func (a *AzureMonitor) Gather(acc telegraf.Accumulator) error {
+	resp, err := a.makeRequest()
+	if err != nil {
+		return err
```

Comment:
Other inputs have leveraged [`acc.AddError`](https://github.com/influxdata/telegraf/search?q=acc.AddError) and returned `nil` from their `Gather()` functions.

May be some usability benefits that come out of this pattern 🤷🏾